### PR TITLE
Pj OBB

### PR DESCRIPTION
**Pull request by Pjbomb2**

### CODE REVIEW

File: include/nbl/asset/utils/IMeshManipulator.h [hunk 1]
```diff
@@ -351,6 +353,211 @@ class IMeshManipulator : public virtual core::IReferenceCounted
...
+			core::vectorSIMDf Min2 = core::vectorSIMDf(Edge10Proj.y, Edge20Proj.y, Edge30Proj.y);
+			core::vectorSIMDf Diff = Max2 - Min2;
+			float Quality = Diff.x * Diff.y + Diff.x * Diff.z + Diff.y * Diff.z;
+			if (Quality < PrevQuality) {
+				PrevQuality = Quality;
+				for (int i = 0; i < 3; i++) {
+					AxesEdge[i] = Edges[i];
+				}
+			}
+		}
```
Comment (devshgraphicsprogramming) [inline include/nbl/asset/utils/IMeshManipulator.h:397]: hide these in the `.cpp` file

File: include/nbl/asset/utils/IMeshManipulator.h [hunk 2]
```diff
@@ -351,6 +353,211 @@ class IMeshManipulator : public virtual core::IReferenceCounted
...
+		static inline core::matrix3x4SIMD calculateOBB(const nbl::asset::ICPUMeshBuffer* meshbuffer) {
```
Comment (devshgraphicsprogramming) [inline include/nbl/asset/utils/IMeshManipulator.h:398]: move implementation to `.cpp` file

File: include/nbl/asset/utils/IMeshManipulator.h [hunk 3]
```diff
@@ -73,6 +73,8 @@ class IMeshManipulator : public virtual core::IReferenceCounted
 			float wage;												//angle wage of the vertex
 			core::vector4df_SIMD position;							//position of the vertex in 3D space
 			core::vector3df_SIMD parentTriangleFaceNormal;			//
+			core::vector3df_SIMD v0s;
+			core::vector3df_SIMD v0t;
```
Comment (devshgraphicsprogramming) [inline include/nbl/asset/utils/IMeshManipulator.h:77]: do not mix PR's together, I know this is for tangents.

Comment (Pjbomb2) [reply include/nbl/asset/utils/IMeshManipulator.h:77]: Heck sorry forgot to delete that bit

File: include/nbl/asset/utils/IMeshManipulator.h (hunk 2 above)
Comment (Pjbomb2) [reply include/nbl/asset/utils/IMeshManipulator.h:398]: so move it all to CMeshManipulator.cpp?

File: include/nbl/asset/utils/IMeshManipulator.h (hunk 1 above)
Comment (AnastaZIuk) [reply include/nbl/asset/utils/IMeshManipulator.h]: resolved

File: include/nbl/asset/utils/IMeshManipulator.h (hunk 2 above)
Comment (AnastaZIuk) [reply include/nbl/asset/utils/IMeshManipulator.h]: resolved

File: include/nbl/asset/utils/IMeshManipulator.h (hunk 3 above)
Comment (AnastaZIuk) [reply include/nbl/asset/utils/IMeshManipulator.h]: resolved

**Pull request merged**